src/backend/distributed/planner/deparse_shard_query.c

```diff
@@ -445,6 +445,16 @@ SetTaskQueryString(Task *task, char *queryString)
 }
 
 
+/*
+ * SetTaskQueryStringList sets the queryStringList of the given task.
+ */
+void
+SetTaskQueryStringList(Task *task, List *queryStringList)
+{
+	task->queryStringList = queryStringList;
```

Comment:
Maybe just add a `SetTaskQueryString(task, StringJoin(queryStringList, ';'));` here and remove the other calls?

Reply:
This makes sense, I realized that normally we dont have ';' at the end of our commands. If they are concatenated we do have ';'. This seems inconsistent. If we use `SetTaskQueryString(task, StringJoin(queryStringList, ';'));` they will be consistent( all will have ';' at the end), the test outputs will change a lot though. Are we okay with this?

Reply:
This function has two callers and they both do `SetTaskQueryString(task, StringJoin(commandList, ';'));` prior to calling this function. If we move that call here, why would it make a difference?

Reply:
`Maybe just add a SetTaskQueryString(task, StringJoin(queryStringList, ';')); here and remove the other calls?`

I thought here you meant we should only use `SetTaskQueryStringList`  and remove all SetTaskQueryString calls since `SetTaskQueryStringList` will call `SetTaskQueryString` inside it. Did I misunderstand you?

Reply:
I guess you meant we should keep the existing calls for `SetTaskQueryString` but instead of calling `SetTaskQueryString` and `SetTaskQueryStringList` we should just call `SetTaskQueryStringList` which will call the former inside it.

Reply:
updated the PR in that way.